website: pgServices Docs (#1881)
graphile/crystal: postgraphile/website/postgraphile/config.mdx

];
```

:::caution Don't set `name` unless you need to!

> We recommend that you don't set a `name` unless you have more than one pgService.
> If you do, the different services may need different connection settings and will
> definitely be listening with different "subscribers" (different events), so the `pgSettingsKey`
> and `pgSubscriberKey` dictate which key to use for this. The default is if the
> `name` is `main` (the default name) then it uses the `pgSettings`/`pgSubscriber` key, otherwise it prefixes this
> with the name of the service and an underscore, for example a pgService with
> `name: "otherdb"` would use the context keys `otherdb_withPgClient`, `otherdb_pgSettings` and `otherdb_pgSubscriber`.

Reviewer: ```suggestion
We recommend that you don't set a `name` unless you have more than one pgService.
If you do, the different services may need different properties on `context` to detail how to connect, the settings to use, and how to subscribe to events. The `withPgClientKey`, `pgSettingsKey`
and `pgSubscriberKey` options dictate under which keys these properties are stored on the context. By default, if the
pgService's `name` is left at its default (`"main"`) then it uses `"withPgClient"`, `"pgSettings"` and `"pgSubscriber"` respectively for these keys; otherwise it prefixes the keys with the name of the service and an underscore, for example a pgService with
`name: "otherdb"` would use the context keys `otherdb_withPgClient`, `otherdb_pgSettings` and `otherdb_pgSubscriber`.
```

:::

> :::warning

Reviewer: This probably renders as "DANGER" right now; we should probably soften that a bit, or make it more useful...

Maybe:

```suggestion
:::warning Don't set `name` unless you need to!
```

Please check this renders correctly and remove the backticks if not.

Reviewer: Might be nice to add this to a few of the other [adminitions](https://docusaurus.io/docs/markdown-features/admonitions) in the documentation too, definitely a separate PR for that though!

> Generally there's no need to set a `name` unless you have more than one pgService.

Reviewer: ```suggestion
We recommend that you don't set a `name` unless you have more than one pgService.
```

When you do, the different services may need different connection settings and will
definitely be listening with different "subscribers" (different events), so the `pgSettingsKey`
and `pgSubscriberKey` dictate which key to use for this. Currently the default is if the
name is `main` (the default name) then we use the `pgSettings`/`pgSubscriber` key, otherwise we prefix this

> with the name of the service and an underscore

Reviewer: ```suggestion
with the name of the service and an underscore, for example a pgService with
`name: "otherdb"` would use the context keys `otherdb_withPgClient`, `otherdb_pgSettings` and `otherdb_pgSubscriber`.
```

Please also integrate `withPgClient`/`withPgClientKey` into the rest of the paragraph, it will definitely change for each pgService too.
